src/Microsoft.OData.Core/ODataMessageWriter.cs

```diff
@@ -842,15 +843,15 @@ private void EnsureODataFormatAndContentType()
 
                     //// TODO: What about the encoding - should we verify that it's 7bit US-ASCII only?
 
-                    this.batchBoundary = ODataBatchWriterUtils.CreateBatchBoundary(this.writingResponse);
+                    this.batchBoundary = ODataMultipartMixedBatchWriterUtils.CreateBatchBoundary(this.writingResponse);
 

```

Comment:
This doesn't seem like it should be in the base ODataMessageWriter class.

Reply:
see above. ODataMessageWriter is a common class for message processing with no derived types.

---
In reply to: [145853276](https://github.com/OData/odata.net/pull/960#discussion_r145853276) [](ancestors = 145853276)

Reply:
See above. batchBoundary should be moved into a multipart-mixed specific class. ODataMessageWriter can call into derived class to get the context through a virtual method whose default implementation is to call HttpUtils.BuildContentType, but is overridden in MultipartMixed to add the batchBoundary type parameter to the content type.

---
In reply to: [146076783](https://github.com/OData/odata.net/pull/960#discussion_r146076783) [](ancestors = 146076783,145853276)